test/testastutils.cpp

```diff
@@ -196,11 +198,11 @@ class TestAstUtils : public TestFixture {
         ASSERT_EQUALS(false, isSameExpression("x < y;", "x", "y"));
         ASSERT_EQUALS(true,  isSameExpression("(x + 1) < (x + 1);", "+", "+"));
         ASSERT_EQUALS(false, isSameExpression("(x + 1) < (x + 1L);", "+", "+"));
-        ASSERT_EQUALS(true,  isSameExpression("(1 + x) < (x + 1);", "+", "+"));
```

Comment:
Does the tokenizer language only depend on "test.cpp" vs. "test.c" or something more?

Reply:
It depends on the file extension unless `Settings::enforceLanguage` is set. See also #5853.

#5725 will get rid of most of the unnecessary language parameters we are passing around as it can be determined from the `Token`.

Reply:
So we could add a parameter to control the language to the macro `isSameExpression()`? The test at the bottom are clearly C++, whereas the rest was probably intended to be C.

Reply:
Yes, that's what we should do. I didn't proceed on that yet.

Reply:
This should work and allows us to keep the existing test coverage.
~~~
#define isSameExpressionC(code, tokStr1, tokStr2) isSameExpression_(__FILE__, __LINE__, code, tokStr1, tokStr2, false)
#define isSameExpressionCPP(code, tokStr1, tokStr2) isSameExpression_(__FILE__, __LINE__, code, tokStr1, tokStr2, true)
    bool isSameExpression_(const char* file, int line, const char code[], const char tokStr1[], const char tokStr2[], bool cpp) {
        const Settings settings;
        Library library;
        Tokenizer tokenizer(&settings, this);
        std::istringstream istr(code);
        ASSERT_LOC(tokenizer.tokenize(istr, cpp ? "test.cpp" : "test.c"), file, line);
        const Token * const tok1 = Token::findsimplematch(tokenizer.tokens(), tokStr1, strlen(tokStr1));
        const Token * const tok2 = Token::findsimplematch(tok1->next(), tokStr2, strlen(tokStr2));
        return (isSameExpression)(cpp, false, tok1, tok2, library, false, true, nullptr);
    }
~~~

Reply:
That was my approach as well - I did apply it a bit different though.

The other tests also need to use C input. But if there is invalid C code an exception is thrown though which is annyoing as you need to use the debugger to figure out which test fails. That needs some (long needed) improvements in the macros. I have started on those locally but haven't finish it up yet.